src/category_theory/sites/dense_subsite.lean

```diff
@@ -485,4 +485,21 @@ begin
     functor_unit_iso_comp' := λ ℱ, by convert α.left_triangle_components }
 end
 
+variables
+  [concrete_category.{max v u} A]
+  [limits.preserves_limits (forget A)]
+  [reflects_isomorphisms (forget A)]
+  [Π (X : C), limits.preserves_colimits_of_shape (J.cover X)ᵒᵖ (forget A)]
+  [∀ (X : C), limits.has_colimits_of_shape (J.cover X)ᵒᵖ A]
+  [Π (X : D), limits.preserves_colimits_of_shape (K.cover X)ᵒᵖ (forget A)]
+  [∀ (X : D), limits.has_colimits_of_shape (K.cover X)ᵒᵖ A]
```

Comment:
Not directly related to this PR, but since this combination is tedious and appears everywhere, does it make sense to define a typeclass `has_sheafification` that extends these?

Or alternatively, `has_sheafification A` might be defined to say that the forgetful `is_right_adjoint (Sheaf_to_presheaf J K A)` for all sites `J` and `K`, and then make an instance out of these, but this might require some more work.